eclipse-theia/theia
# dep: update `markdown-it` dependency

#### What it does


The pull-request addresses a security vulnerability ([`Uncontrolled Resource Consumption in markdown-it
`](https://github.com/advisories/GHSA-6vfc-qv3f-vr6c)) with `markdown-it`. The change also makes `markdown-it` a shared dependency (exported by `@theia/core`) since it is used by many packages, and they no longer need to reference it themselves at different versions.

#### How to test


1. confirm that `yarn audit --level=moderate` does not produce any output for `markdown-it`.
2. confirm that markdown rendering works for the preferences-ui, vsx-registry extensions tooltip, etc.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>
